service/storage_proxy.cc

```diff
@@ -3296,6 +3308,25 @@ storage_proxy::create_write_response_handler(const std::tuple<lw_shared_ptr<paxo
                     inet_address_vector_topology_change(), inet_address_vector_topology_change(), std::move(tr_state), get_stats(), std::move(permit), std::monostate(), is_cancellable::no);
 }
 
+// After sending the write to replicas(targets), the replicas might generate send view updates. If the number of view updates
+// that the replica is already sending is too high, the replica should not send any more view updates, and consequently,
+// we should not send the write to the replica.
+template<typename Range>
+bool storage_proxy::can_update_views(const Range& targets, const schema_ptr& s, db::write_type type, tracing::trace_state_ptr tr_state) const {
+    static constexpr double mv_admission_control_threshold = 0.8;
+    if (type == db::write_type::VIEW) {
+        return true;
+    }
+    if (!_db.local().find_column_family(s).views().empty() && targets.begin() != targets.end()) {
+        for (auto ep : targets) {
+            if (get_backlog_of(ep).relative_size() >= mv_admission_control_threshold) {
+                return false;
+            }
+        }
+    }
```

Comment:
Could also be written with std::ranges::all_of(targets, ...)

Reply:
I used this in the rebase